core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -253,8 +254,42 @@ public static TableMetadata read(FileIO io, InputFile file) {
     }
   }
 
-  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  /**
+   * Read TableMetadata from a JSON string.
+   * <p>
+   * The TableMetadata's metadata file location will be unset.
+   *
+   * @param io a FileIO used by {@link Snapshot} instances
+   * @param json a JSON string of table metadata
+   * @return a TableMetadata object
+   */
+  public static TableMetadata fromJson(FileIO io, String json) {
+    return fromJson(io, null, json);
```

Comment:
doesn't this `null` cause NPE on line 288?

Reply:
No, that method expects `JsonNode`, not a String. This is calling `fromJson(FileIO, String, String)`.

Reply:
This is also validated by the tests, which call this new method quite a bit.

Reply:
ok, IDE must have been confused because it was traversing differently (I was also getting ambiguous method references).